kani-compiler/src/codegen_cprover_gotoc/codegen/function.rs

```diff
@@ -232,11 +233,14 @@ impl<'tcx> GotocCtx<'tcx> {
         self.reset_current_fn();
     }
 
-    pub fn is_proof_harness(&self, def_id: DefId) -> bool {
-        let all_attributes = self.tcx.get_attrs_unchecked(def_id);
-        let (proof_attributes, _) = partition_kanitool_attributes(all_attributes);
+    /// Check that if an item is tagged with a proof_attribute, it is a valid harness.
+    fn check_proof_attribute(&self, def_id: DefId, proof_attributes: Vec<&Attribute>) {
         if !proof_attributes.is_empty() {
```

Comment:
This condition is also checked by the callers. Should we remove it and turn it into an assert?

Reply:
Sure